### PR TITLE
fix remaining field selectors 

### PR DESCRIPTION
**Pull request by deads2k**
Really hoping that https://github.com/openshift/origin/pull/16305 works out.  This updates all the rest for that pattern and allows us to remove a ton of boilerplate while maintaining decent unit test coverage.  Still need a "you forgot to add a test" test, but that was already missing.

### CODE REVIEW

File: pkg/authorization/apis/authorization/v1/conversion.go [hunk 1]
```diff
@@ -394,15 +394,32 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
...
+
+func addLegacyFieldSelectorKeyConversions(scheme *runtime.Scheme) error {
+	if err := scheme.AddFieldLabelConversionFunc(LegacySchemeGroupVersion.String(), "PolicyBinding", legacyPolicyBindingFieldSelectorKeyConversionFunc); err != nil {
 		return err
 	}
+	return nil
+}
 
+func addFieldSelectorKeyConversions(scheme *runtime.Scheme) error {
 	return nil
```
Comment (soltysh) [inline pkg/authorization/apis/authorization/v1/conversion.go]: Why this is empty?

File: pkg/build/apis/build/v1/conversion.go [hunk 2]
```diff
@@ -174,24 +174,45 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
...
+		return err
+	}
+	return nil
+}
 
-	if err := scheme.AddFieldLabelConversionFunc("v1", "BuildConfig", apihelpers.LegacyMetaV1FieldSelectorConversionWithName); err != nil {
+func addFieldSelectorKeyConversions(scheme *runtime.Scheme) error {
+	if err := scheme.AddFieldLabelConversionFunc(SchemeGroupVersion.String(), "Build", buildFieldSelectorKeyConversionFunc); err != nil {
 		return err
 	}
```
Comment (soltysh) [inline pkg/build/apis/build/v1/conversion.go]: No BC here?

File: pkg/user/apis/user/v1/conversion.go [hunk 3]
```diff
@@ -2,17 +2,49 @@ package v1
...
+
+func legacyIdentityFieldSelectorKeyConversionFunc(label, value string) (internalLabel, internalValue string, err error) {
+	switch label {
+	case "providerName",
+		"providerUserName",
+		"user.name",
+		"user.uid":
+		return label, value, nil
+	default:
+		return runtime.DefaultMetaV1FieldSelectorConversion(label, value)
```
Comment (soltysh) [inline pkg/user/apis/user/v1/conversion.go]: `return apihelpers.LegacyMetaV1FieldSelectorConversionWithName(label, value)`?

File: pkg/authorization/apis/authorization/v1/conversion.go (hunk 1 above)
Comment (deads2k) [reply pkg/authorization/apis/authorization/v1/conversion.go]: > Why this is empty?

We don't have any custom field selectors.  Yet.  I'm placing the skeleton here so people can more easily figure out what to do.  This isn't really obvious.

File: pkg/build/apis/build/v1/conversion.go (hunk 2 above)
Comment (deads2k) [reply pkg/build/apis/build/v1/conversion.go]: > No BC here?

No.  BC only supports metadata, but we have the "name" -> "metadata.name" mapping for the ungroupified versions.

File: pkg/user/apis/user/v1/conversion.go (hunk 3 above)
Comment (deads2k) [reply pkg/user/apis/user/v1/conversion.go]: > return apihelpers.LegacyMetaV1FieldSelectorConversionWithName(label, value)?

This one never had the "name" -> "metadata.name" mapping.

File: pkg/image/apis/image/fields.go [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+func ImageStreamSelector(obj runtime.Object, fieldSet fields.Set) error {
+	imageStream, ok := obj.(*ImageStream)
+	if !ok {
+		return fmt.Errorf("%T not an ImageStream", obj)
 	}
+	fieldSet["spec.dockerImageRepository"] = imageStream.Spec.DockerImageRepository
+	fieldSet["status.dockerImageRepository"] = imageStream.Status.DockerImageRepository
```
Comment (mfojtik) [inline pkg/image/apis/image/fields.go]: @deads2k we now also have status.publicDockerImageRepository if that matters

Comment (deads2k) [reply pkg/image/apis/image/fields.go]: > @deads2k we now also have status.publicDockerImageRepository if that matters

Doesn't look like it's used as a field selector.

**Pull request merged**